packages/core/src/external/aws/lambda-logic/bundle-to-html-bmi.ts

```diff
@@ -1632,6 +1646,36 @@ function createObservationLaboratorySection(
   return createSection("Laboratory", `${panelContent}`);
 }
 
+function findPanelsWithHba1c(
+  diagnosticReports: DiagnosticReport[],
+  allObservations: Observation[],
+  excludedPanels: Set<DiagnosticReport>
+): DiagnosticReport[] {
+  const a1cLoincCode = "4548-4";
+
+  return diagnosticReports
+    .filter(report => !excludedPanels.has(report))
+    .filter(report => {
+      if (!report.result) return false;
+
+      return report.result.some(reference => {
+        const observationId = reference.reference?.split("/")[1];
+        const observation = allObservations.find(obs => obs.id === observationId);
+
+        return observation?.code?.coding?.some(coding => {
+          const code = fetchCodingCodeOrDisplayOrSystem(coding, "code");
+          return code === a1cLoincCode;
+        });
+      });
+    })
+    .sort((a, b) => {
+      const dateA = a.effectiveDateTime || a.effectivePeriod?.start || "";
+      const dateB = b.effectiveDateTime || b.effectivePeriod?.start || "";
+      return dayjs(dateB).diff(dayjs(dateA));
+    })
+    .slice(0, 2);
```

Comment:
why slice 0,2 ?

Reply:
Thats how they wanted it - they want to limit each type to 2 and limit 4 overall

Reply:
sg, seems like a gotcha if this fn gets called again, can you instead slice at the callsite? The assumption of this function would usually be that it finds all panels.
```
findPanelsWithHba1c(diagnosticReports, observations, includedPanels).slice(0,2)
```

Reply:
good call out - updated